### DISCUSSION TITLE
Fingers joints move of yarp icub through python

### DISCUSSION THREAD

**Discussion by ArfnZl** [robotology/community#572] [Q&A] [answered]
Please provide python yarp icub code to move icub fingers joints move....i want to move fingers joint in yarp icub simulator through python code

**Comment by traversaro** [comment, marked as answer]
Hi @ArfnZl . The fingers are exposed using the same controlboard mechanism used to move all the other joints of the robot. 
You can find an example on how to move the position of the shoulder (i.e. axis 0-3) of the arm via Python in https://github.com/robotology/icub-tutorials/blob/v1.21.0/python/python-motor-control.py . The fingers are controlled by the axis 9-15 of the same controlboard. I would suggest that you get familiar with the motion of each axis using the `yarpmotorgui`, and then once you understand the motion connected at each axis, you can modify the Python snippet to move the fingers that you want to move.

**Comment by ArfnZl** [comment]
I am facing  another problem...i have added tactile  sensors on icub skin
arm and they are giving  values now i want to add pressure  tactile
sensors that give value according  to the object..and when i grip object it
give values according  to object whether  it is hard or soft object

**Reply by pattacini** [reply]
This is a completely different question @ArfnZl that I would mark as off-topic.
Could you please open up a dedicated discussion for that?
Thanks in advance.
